src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,155 @@ static void Main(string[] args)
                 Diagnostic(ErrorCode.ERR_BadArity, "Goo<>").WithArguments("Program.Goo<T, U>(T, U)", "method", "2"));
         }
 
+        [Fact]
+        public void Bug50782_2()
+        {
+            string source = @"
+#nullable enable
+
+interface IOperation<T> { }
+
+public class StringOperation : IOperation<string?> { }
+
+public class C {   
+    static void Main() {
+        TestF(out var _);
+        TestF(out _);
+    }
+   
+    static void TestF(out string result) => result = """";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics();
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree);
+
+            // out _
+            ForEach(GetDiscardIdentifiers(tree), (IdentifierNameSyntax discardOut) => {
+                Assert.Null(model.GetDeclaredSymbol(discardOut));
+                var discardSymbol = (IDiscardSymbol)model.GetSymbolInfo(discardOut).Symbol;
+                Assert.Equal("System.String", discardSymbol.Type.ToTestDisplayString());
+                Assert.Equal("System.String", model.GetTypeInfo(discardOut).Type.ToTestDisplayString());
+            });
+
+
+            // out T _, out var _, out T? _
```

Comment:
nit: this comment is confusing since the source code only include `out var _` case